assessment-api/ddl.sql

```diff
@@ -0,0 +1,227 @@
+  updated_by            varchar(2000),
+  updated_at            timestamp);
+ 
+-- Create table assessment taxons;
+
+create table assessment_taxons
+( assessment_taxon_id        integer       primary key,
+  assessment_tool_id         integer       not null,
+  parent_assessment_taxon_id integer,
+  assessment_taxon_name      varchar(50),
+  permalink                  varchar(500),
+  assessment_taxon_descr     varchar(2000),
+  created_by                 varchar(2000) not null,
+  created_at                 timestamp     not null,
+  updated_by                 varchar(2000),
+  updated_at                 timestamp);
+ 
+-- Create table taxon_dimensions
+
+create table assessment_taxon_dimensions
+( assessment_taxon_dimension_id serial        primary key,
+  assessment_taxon_id           integer       not null,
+  dimension_name                varchar(30)   not null,
+  created_by                    varchar(2000) not null,
+  created_at                    timestamp     not null,
+  updated_by                    varchar(2000),
+  updated_at                    timestamp);
+
+
+-- Create table requirements (products & services)
+
+create table requirements
+( requirement_id    serial        primary key,
+  requirement_name  varchar(100)  not null,
+  requirement_descr varchar(2000) not null,
+  created_by        varchar(2000) not null,
+  created_at        timestamp     not null,
+  updated_by        varchar(2000),
+  updated_at        timestamp);
+
+ 
+-- Create table requirement_taxons
+
+create table requirement_taxons
+( requirement_taxon_id serial        primary key,
+  requirement_id       integer       not null,
+  assessment_taxon_id  integer       not null,
+  created_by           varchar(2000) not null,
+  created_at           timestamp     not null,
+  updated_by           varchar(2000),
+  updated_at           timestamp);
+
+  
+
+-- Create table lot_requirement_taxons
+
+create table lot_requirement_taxons
+( lot_requirement_taxon_id serial        primary key,
+  lot_id               integer       not null,
+  requirement_taxon_id integer       not null,
+  created_by           varchar(2000) not null,
+  created_at           timestamp     not null,
+  updated_by           varchar(2000),
+  updated_at           timestamp);
+
+
+-- Create table supplier_submissions 
+  
+create table supplier_submissions
+( supplier_submission_id        serial        primary key,
+  lot_requirement_taxon_id      integer       not null,
+  supplier_id                   integer       not null,
+  assessment_submission_type_id integer       not null,
+  submission_reference          varchar(30),
+  submission_value              integer,
+  created_by                    varchar(2000) not null,
+  created_at                    timestamp     not null,
+  updated_by                    varchar(2000),
+  updated_at                    timestamp);
+
+
+-- Create table Assessments
+
+create table assessments(
+  assessment_id         integer       primary key,
```

Comment:
Should this be `BIGSERIAL PRIMARY KEY` (or similar) - so id is autogenerated on insert? (just trying to implement the `create assessment` API endpoint)

Reply:
I think serial will be good enough. I am only expect n suppliers for each agreement making 2 returns. It is possible they may make 10. However I doubt it. Hapy to make it BIGSERIAL if you prefer.

Reply:
Serial is preferable thanks - less knock on changes in the code